### PR TITLE
Enable different encoder and decoder sizes in cross-attention

### PR DESCRIPTION
**Pull request by avidale**
**What does this PR do? Please describe:**

This PR allows creating an attention layer with different dimensions of the inputs and outputs. 
This is useful for encoder-decoder architectures with different dimensionalities (including low-dimensional SONAR encoders and decoders, as an example). 

**Does your PR introduce any breaking changes? If yes, please list them:**
No breaking changes. 

**Check list:**
- [ ] Was the content of this PR **discussed and approved** via a GitHub issue? (no need for typos or documentation improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairseq2/blob/main/CONTRIBUTING.md)?
- [x] Did you make sure that your **PR does only one thing** instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**?
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you **update the [CHANGELOG](https://github.com/facebookresearch/fairseq2/blob/main/CHANGELOG.md)**? (no need for typos, documentation, or minor internal changes)

### CODE REVIEW

File: src/fairseq2/nn/transformer/multihead_attention.py [hunk 1]
```diff
@@ -187,6 +187,7 @@ def __init__(
         model_dim: int,
         num_heads: int,
         *,
+        input_dim: Optional[int] = None,
```
Comment (cbalioglu) [inline src/fairseq2/nn/transformer/multihead_attention.py:190]: I don't think `input_dim` is the right name for this parameter. Pedantically "input" to MHA is the query. K/V values are more like the "state dictionary" to attend against. How about naming this `kv_model_dim` or `kv_dim`?

**Review by cbalioglu — COMMENTED** [review]
Thanks for the PR and including CHANGELOG and tests as well. Just left a rather nit comment about the naming of this parameter. Once we align on that, I think it will be good to go.

File: src/fairseq2/nn/transformer/multihead_attention.py (hunk 1 above)
Comment (avidale) [reply src/fairseq2/nn/transformer/multihead_attention.py:190]: `kv_dim` sounds just right!

**Review by cbalioglu — APPROVED** [review]
Nice! Thanks again for your contribution! Looking forward to future ones! :)

**Pull request merged**